rust/cardano-chain-follower/src/mithril_turbo_downloader.rs

```diff
@@ -304,52 +305,43 @@ impl MithrilTurboDownloader {
     }
 
     /// Parallel Download, Extract and Dedup the Mithril Archive.
-    async fn dl_and_dedup(&self, location: &str, target_dir: &Path) -> MithrilResult<()> {
+    async fn dl_and_dedup(&self, location: &str) -> anyhow::Result<()> {
         // Get a copy of the inner data to use in the sync download task.
         let inner = self.inner.clone();
-        let location = location.to_owned();
-        let target_dir = target_dir.to_owned();
+
+        debug!("Download Snapshot location='{location}'.");
+        let dl_config = self.inner.cfg.dl_config.clone().unwrap_or_default();
+        let dl_processor =
+            ParallelDownloadProcessor::new(location, dl_config, inner.cfg.chain).await?;
+
+        // Decompress and extract and de-dupe each file in the archive.
+        stats::mithril_extract_started(inner.cfg.chain);
+
+        // We also immediately start downloading now.
+        stats::mithril_dl_started(inner.cfg.chain);
+
+        // Save the DownloadProcessor in the inner struct for use to process the downloaded data.
+        self.inner
+            .dl_handler
+            .set(dl_processor)
+            .map_err(|_| anyhow!("Failed to set the inner dl_handler. Must already be set?"))?;
 
         // This is fully synchronous IO, so do it on a sync thread.
-        let result = spawn_blocking(move || {
```

Comment:
This was deferred because stats must always be updated doesn't matter why it ended.  The stats record update is wrong below.  Needs to be before result is checked as noted below.

So, its not possible to do map_err.  change it back how it was and move where `stats::mithril_dl_finished(self.inner.cfg.chain, None);` is called to immediately after the `await()`